NuclearData/mgNeutronData/baseMgNeutron/baseMgNeutronDatabase_class.f90

```diff
@@ -290,12 +298,22 @@ subroutine init(self, dict, ptr, silent)
 
     ! Load and verify number of groups
     self % nG = self % mats(1) % nGroups()
-    do i=2,nMat
+    do i = 2,nMat
       if(self % nG /= self % mats(i) % nGroups()) then
         call fatalError(Here,'Inconsistant # of groups in materials in matIdx'//numToChar(i))
       end if
     end do
 
+    ! Precalculate majorant xs for delta tracking
+    allocate (self % majorant(self % nG))
+    do g = 1,self % nG
+      xs = ZERO
+      do i = 1,nMat
+        xs = max(xs, self % mats(i) % data(TOTAL_XS, g))
+      end do
+      self % majorant(g) = xs
```

Comment:
```suggestion
      self % majorant(g) = maxval(self % mats(:) % data(TOTAL_XS, g)
```
But please check that it really does work. My Fortran is a bit rusty.

Reply:
True Fortran magic